refactor MoviesController: Use repository instead of JSON

lo1d1 opened the pull request:
Switches data retrieval from a static JSON file to the IMoviesRepository. This decouples the controller from the file system and ensures data is fetched consistently from the database.

lo1d1 committed:
refactor MoviesController: Use repository instead of JSON

lo1d1 committed:
Add unit tests for MovieController

lo1d1 committed:
Change old Integration test for new MovieController

01vadim10 committed:
Merge branch 'main' into refactor/movies-controller

01vadim10 commented on tests/SneakPeek.IntegrationTests/MoviesControllerTests.cs:
```diff
@@ -1,38 +1,60 @@
-namespace SneakPeak.IntegrationTests;
-
-public class MoviesControllerTests : IClassFixture<WebApplicationFactory<Program>>
+public class MovieControllerTest : IClassFixture<WebApplicationFactory<Program>>
```
Why did you change the name of the controller?

01vadim10 committed:
Merge branch 'main' into refactor/movies-controller

01vadim10 committed:
Merge branch 'main' of https://github.com/01vadim10/SneakPeek into refactor/movies-controller

01vadim10 committed:
Merge branch 'main' into refactor/movies-controller

01vadim10 committed:
Merge branch 'main' into refactor/movies-controller

01vadim10 committed:
Fix CodeQL generic catch clause and test warnings

- Remove generic Exception catch in MoviesController to let framework handle errors
- Replace Assert.Equal(null, result) with Assert.Null(result) per xUnit best practices
- Add null-forgiving operator for AddMovieAsync(null!) test

01vadim10 committed:
Trigger CodeQL re-scan

01vadim10 committed:
Replace Moq with NSubstitute and xUnit assertions with Shouldly

- Replace Moq with NSubstitute in unit tests
- Replace xUnit assertions with Shouldly for better readability
- Fix test class naming: MovieControllerTest/Tests -> MoviesControllerTests
- Add test coverage for NotFound scenarios (null and empty list)
- Remove unused Moq dependency from integration tests

Addresses code review feedback for PR #36

01vadim10 committed:
Fix CodeQL useless upcast warning in null test

Remove unnecessary explicit cast by using a typed variable instead.
This resolves the CodeQL warning about upcast from null.
